lldb/source/Target/TargetList.cpp

```diff
@@ -263,8 +267,9 @@ Status TargetList::CreateTargetInternal(
   // If we have a valid architecture, make sure the current platform is
   // compatible with that architecture
   if (!prefer_platform_arch && arch.IsValid()) {
-    if (!platform_sp->IsCompatibleArchitecture(arch, false, &platform_arch)) {
-      platform_sp = Platform::GetPlatformForArchitecture(arch, &platform_arch);
+    ArchSpec compatible_arch;
+    if (!platform_sp->IsCompatibleArchitecture(arch, false, &compatible_arch)) {
+      platform_sp = Platform::GetPlatformForArchitecture(arch, nullptr);
```

Comment:
What does this part of the patch achieve? I can see the other part makes fat binaries behave as non-fat ones, but I don't understand this part.

Reply:
`platform_sp->IsCompatibleArchitecture(arch, false, &compatible_arch))`
 is supposed to be query, but even if it fails it overwrites the out-param, thus erasing platform_architecture.

Reply:
Bu tit seems like even if it succeeds we don't change `platform_arch`. Should we?

Reply:
Fixed. Thanks for noticing!!